pkg/server/server.go

```diff
@@ -237,6 +244,15 @@ func New(
 
 		if v := tuning.Snapshot; v != nil {
 			logrus.WithFields(logrus.Fields{"threshold": v.Threshold, "trailing": v.Trailing}).Print("Configure dqlite raft snapshot parameters")
+			if v.Trailing < v.Threshold {
+				return nil, fmt.Errorf("the snapshot's trailing parameter must be less than the threshold parameter")
+			}
+			// ensure that the threshold is at least 384 and trailing is at least 512 to avoid CPU utilization issues.
+			if v.Threshold < min_trailing || v.Trailing < min_trailing {
```

Comment:
I think the two conditions should be separated. Otherwise we would overwrite the requested behaviour without reason. For example imagine something like:

```yaml
threshold : 200
trailing: 1024
```

This logic would use `trailing = 512` even if the user-provided value is ok.